src/main.rs

```diff
@@ -118,15 +118,15 @@ fn main() {
         }
         Commands::CheckHash { giant_uri, hash } => {
             let mut client = GiantApiClient::new(giant_uri.clone());
-            CliResult::new(client.check_hash_exists(hash), FailureExitCode::Api)
+            CliResult::new(client.check_hash_exists(hash).await, FailureExitCode::Api)
                 .print_or_exit(format);
         }
         Commands::CheckFile { giant_uri, path } => {
-            let mut client = GiantApiClient::new(giant_uri.clone());
-            let file_exists = (|| {
+            let file_exists = (|| async {
+                let mut client = GiantApiClient::new(giant_uri.clone());
                 let hash = hash_file(path.clone())?;
-                client.check_hash_exists(&hash.hash)
-            })();
+                client.check_hash_exists(&hash.hash).await
+            })().await;
```

Comment:
I'm trying to figure out why I used an immediately invoked function here... You didn't by any chance try to do it without the function? Something like

```rust
let mut client = GiantApiClient::new(giant_uri.clone());
let hash = hash_file(path.clone())?;
let file_exists = client.check_hash_exists(&hash.hash).await;
```

Reply:
It's to capture the `Result<bool, CliError>` in one place so it can be passed to `CliResult::new`.

You can't use the `?` in `hash_file(path.clone())?` otherwise because it could exit early on that line and your command wants a return type of `CliResult`, not `Result`

Maybe there's a better way of doing it though.